src/server.go

```diff
@@ -76,9 +77,38 @@ func (a *App) QueryHandler(key []byte, w http.ResponseWriter, r *http.Request) {
   }
 }
 
+func (a *App) GetAuthorization(w http.ResponseWriter, r *http.Request) (string, bool) {
+  var authstring string
+  if a.basicauth == true {
+    authtoken := r.Header.Get("Authorization")
+    if authtoken != "" {
+      authstringb, _ := base64.StdEncoding.DecodeString(strings.Split(authtoken, "Basic ")[1])
+      authstring = string(authstringb)
+      authstring = authstring+"@"
+      remote := fmt.Sprintf("http://%s%s", authstring, a.volumes[rand.Intn(len(a.volumes))])
+      resp, err := http.Get(remote)
+      if err != nil {
```

Comment:
Merge these? Why pass through the status code?

Reply:
How do you suggest the merge? If there was an error there is no status code, e.g., error we send out 500; no error we send out nginx status code (such as 401)

Reply:
Sure, but I don't like that w and r are being passed in here. Principle of minimal scope.

authtoken := r.Header.Get("Authorization") belongs outside
And I think it should always be a 500 if the volume server is down, so w.WriteHeader(500) belongs on line 115